spec/features/idv/doc_auth/socure_document_capture_spec.rb

```diff
@@ -87,6 +87,99 @@
+          socure_docv_send_webhook(
+            docv_transaction_token: @docv_transaction_token,
+            event_type: 'DOCUMENT_FRONT_UPLOADED',
+          )
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+          visit idv_socure_document_capture_path
+          expect(page).to have_current_path(idv_socure_document_capture_path)
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+        end
+
+        it 'decline TOS making session expire, generates new capture app' do
+          document_capture_session = DocumentCaptureSession.find_by(user_id: @user.id)
+          expect(page).to have_current_path(fake_socure_document_capture_app_url)
+          visit idv_socure_document_capture_path
+          expect(page).to have_current_path(idv_socure_document_capture_path)
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+          socure_docv_send_webhook(
+            docv_transaction_token: @docv_transaction_token,
+            event_type: 'SESSION_EXPIRED',
+          )
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).to be_nil
+          visit idv_socure_document_capture_path
+          expect(page).to have_current_path(idv_socure_document_capture_path)
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+        end
+        it 'cancel idv making session complete, generates new capture app' do
+          document_capture_session = DocumentCaptureSession.find_by(user_id: @user.id)
+          expect(page).to have_current_path(fake_socure_document_capture_app_url)
+          visit idv_socure_document_capture_path
+          expect(page).to have_current_path(idv_socure_document_capture_path)
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+          socure_docv_send_webhook(
+            docv_transaction_token: @docv_transaction_token,
+            event_type: 'SESSION_COMPLETE',
+          )
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).to be_nil
+          visit idv_socure_document_capture_path
+          expect(page).to have_current_path(idv_socure_document_capture_path)
+          document_capture_session.reload
+          expect(document_capture_session.socure_docv_capture_app_url).
+            to eq(fake_socure_document_capture_app_url)
+        end
```

Comment:
canceling idv is actioned by clicking the cancel button and confirming
![Screenshot 2024-11-27 at 1 58 43 PM](https://github.com/user-attachments/assets/4b6d2d77-2030-453a-925a-a764149dfdf4)
when restarting idv the user gets a new document capture session